contactsummary/js/src/contact-summary/services/contract.service.js

```diff
@@ -14,52 +14,79 @@ define([
-    };
+    return _.assign(factory, {
+      get: get,
+      getCollection: getCollection,
+      getContracts: getContracts,
+      getContractDetails: getContractDetails,
+      getLengthOfService: getLengthOfService,
+      getPrimary: getPrimary,
+      resetContracts: resetContracts
+    });
 
-    /**
-     * @returns {*}
-     */
-    factory.get = function () {
-      /** @type {(contractService|ModelService)} */
-      var self = this;
+    function assembleContracts () {
+      var deferred = $q.defer();
+      var promises = [];
 
-      return init().then(function () {
-        // return self.getData();
-        return self.getCollection();
+      angular.forEach(contracts, function (contract) {
+        var assembledContract = {};
+
+        assembledContract.id = contract.id;
+        assembledContract.is_primary = contract.is_primary;
+        assembledContract.is_current = contract.is_current;
+        assembledContract.revision_id = null;
+
+        if (contract.api_HRJobContractRevision_getcurrentrevision) {
+          assembledContract.revision_id = contract.api_HRJobContractRevision_getcurrentrevision.values.id;
+        }
+
+        var promise = factory.getContractDetails(contract.id)
+          .then(function (response) {
+            assembledContract.title = response.title;
+            assembledContract.start_date = response.period_start_date;
+            assembledContract.end_date = response.period_end_date;
+            assembledContract.type = response.contract_type;
+            assembledContract.pay = response.pay;
+            assembledContract.hours = response.hours;
+          })
+          .then(function () {
+            factory.collection.insertItem(contract.id, assembledContract);
+          });
+
+        promises.push(promise);
       });
-    };
 
-    /**
-     * A primary contract is:
-     * 1. (If exists) a contract with is_primary=1 that is active, or
-     * 2. The most recent contract that is active
-     */
-    factory.getPrimary = function () {
-      return this.get().then(function (response) {
-        var sortedContracts = _.sortBy(response, function (o) {
-          return [o.end_date, +o.is_primary];
+      $q.all(promises)
+        .catch(function (response) {
+          $log.error('Something went wrong', response);
+        })
+        .finally(function () {
+          deferred.resolve();
         });
 
-        return _.last(sortedContracts) || {};
-      });
-    };
+      return deferred.promise;
+    }
 
     /**
-     * Reset contracts and promiseCache to initial state
+     * @returns {*}
      */
-    factory.resetContracts = function () {
-      contracts = [];
-      promiseCache = {};
-      initializeCollection();
-    };
+    function get () {
+      /** @type {(contractService|ModelService)} */
+      return init().then(function () {
+        return factory.getCollection();
+      });
+    }
 
-    factory.getContracts = function () {
+    function getCollection () {
```

Comment:
Return is not documented, same in below function.

Reply:
honestly i'm not going to fix the code itself, i'm just fixing the structure here. Otherwise this PR would take weeks

Reply:
Sure no problem, since you were adding some comments, I pointed this out.